#! /bin/sh -e

# DP: hotshot/pstats.py: Error out on missing profile/pstats modules.
# DP: Add a note to the library documentation, that the profile and pstats
# DP: modules can be found in the python2.x-profiler package.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Doc/lib/libprofile.tex~	2003-10-20 16:34:44.000000000 +0200
+++ Doc/lib/libprofile.tex	2005-02-08 23:49:35.000000000 +0100
@@ -2,6 +2,11 @@
 
 \sectionauthor{James Roskind}{}
 
+Debian note: The license for the \module{profile} and \module{pstats}
+modules doesn't conform to the Debian Free Software Guidelines (DFSG).
+These modules can be found in the \code{python-profiler} package in
+the \code{multiverse} section of the Ubuntu archives.
+
 Copyright \copyright{} 1994, by InfoSeek Corporation, all rights reserved.
 \index{InfoSeek Corporation}
 
@@ -264,6 +269,11 @@
 \modulesynopsis{Python profiler}
 
 
+Debian note: The license for the \module{profile} and \module{pstats}
+modules doesn't conform to the Debian Free Software Guidelines (DFSG).
+These modules can be found in the \code{python-profiler} package in
+the \code{multiverse} section of the Ubuntu archives.
+
 
 The primary entry point for the profiler is the global function
 \function{profile.run()}.  It is typically used to create any profile
